### DISCUSSION TITLE
Ended because API returned an empty page

### DISCUSSION THREAD

**Discussion by JohnnyLax21** [victornpb/undiscord#468] [1. Questions & Answers] [answered]
Same issue that seems to have popped up quite a bit before. I can only get it to delete 30-50 messages at a time before having to restart the process. Anything I can do to fix it? 
I know the discord API was having some issues in the past but at least according to the internal tool, it's fine right now. 

<img width="554" alt="Screen Shot 2023-03-24 at 7 36 03 AM" src="https://user-images.githubusercontent.com/52734434/227511131-9ec0a588-1ca6-4fec-9629-c11ef703b4e6.png">

**Comment by Klownicle** [comment]
Dealing with this as well,

Started at 3/24/2023, 8:28:50 AM
<!removed!> minId = ""	maxId = ""	hasLink = false	hasFile = true
Fetching messages...
Grand total: 124	(Messages in current page: 24	To be deleted: 24	Skipped: 0)	offset: 0
Delete delay: 1050ms, Search delay: 2150ms	Last Ping: 261ms, Average Ping: 245ms
Rate Limited: 260 times.	Total time throttled: 0h 2m 24s.
Estimated time remaining: 0h 2m 51s
Waiting for your confirmation...
OK
Fetching messages...
Grand total: 124	(Messages in current page: 8	To be deleted: 8	Skipped: 0)	offset: 0
Delete delay: 1050ms, Search delay: 2150ms	Last Ping: 488ms, Average Ping: 300ms
Rate Limited: 260 times.	Total time throttled: 0h 2m 24s.
Estimated time remaining: 0h 2m 58s
Fetching messages...
Grand total: 124	(Messages in current page: 0	To be deleted: 0	Skipped: 0)	offset: 0
Delete delay: 1050ms, Search delay: 2150ms	Last Ping: 152ms, Average Ping: 245ms
Rate Limited: 260 times.	Total time throttled: 0h 2m 24s.
Estimated time remaining: 0h 2m 51s
Ended because API returned an empty page.
[End condition A].	{"running":true,"delCount":32,"failCount":0,"grandTotal":124,"offset":0,"iterations":3,"_seachResponse":{"analytics_id":"<!removed!>","doing_deep_historical_index":false,"total_results":124,"messages":[]},"_messagesToDelete":[],"_skippedMessages":[]}
Ended at 3/24/2023, 8:29:36 AM! Total time: 0h 0m 46s
Delete delay: 1050ms, Search delay: 2150ms	Last Ping: 152ms, Average Ping: 245ms
Rate Limited: 260 times.	Total time throttled: 0h 2m 24s.
Deleted 32 messages, 0 failed.

**Comment by curlyfries1111** [comment]
I'm also dealing with the same thing, thousands of my deleted messages are also reappearing

**Reply by Klownicle** [reply]
No reappearing messages for me that I've noticed.

**Reply by JohnnyLax21** [reply]
I haven't noticed any of my messages reappearing so far, just the tool not playing nice

**Comment by PetrBlue** [comment]
Same!

**Comment by wwincast** [comment]
is there any fix to this yet?

**Comment by victornpb** [comment, marked as answer]
This issue is currently being handled here #471
